lib/datasource/remote/model/token_model.dart

```diff
@@ -113,8 +117,37 @@ class TokenModel extends Equatable {
   @override
   List<Object?> get props => [chainName, contract, symbol];
 
-  String getAssetString(double quantity) {
-    return "${quantity.toStringAsFixed(precision)} $symbol";
+  static String getAssetString(String? id, double quantity) {
+    if (id!=null && TokenModel.fromId(id)!=null && contractPrecisions.containsKey(id)) {
+      final symbol = TokenModel.fromId(id)!.symbol;
+      return symbol==null ? "" : "${quantity.toStringAsFixed(contractPrecisions[id]!)} $symbol";
+    } else {
+      return "";
+    }
+  }
+
+  void setPrecisionFromString(String s) {
```

Comment:
Why is there a static table? Why not store a list of tokens and each token has all attributes it needs to display, ie, precision. 

Also precision is not based on the contract, it's based on the issued token as far as I know. Meaning a contract can have tokens with different precisions.

Reply:
The whole precision business is schizophrenic because what you (usually) want to show the user is not necessarily the issued token precision. Lots of clumsy code related to this (IIRC it was my first few weeks of Dart programming) but it's been tested, it works, and has been running in the field.

Reply:
Yeah but needs to be stored in the token, no? Not in some other table?